packages/vite/src/node/plugins/worker.ts

```diff
@@ -77,41 +88,52 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
           // inline as blob data url
           return `const encodedJs = "${content.toString('base64')}";
             const blob = typeof window !== "undefined" && window.Blob && new Blob([atob(encodedJs)], { type: "text/javascript;charset=utf-8" });
-            export default function WorkerWrapper() {
+            export default function WorkerWrapper(workerOptions) {
               const objURL = blob && (window.URL || window.webkitURL).createObjectURL(blob);
               try {
-                return objURL ? new Worker(objURL) : new Worker("data:application/javascript;base64," + encodedJs, {type: "module"});
```

Comment:
ping @andreasg123 

I don't quite understand why we didn't pass `{type: "module"}` to `new Worker(objURL)` (https://github.com/vitejs/vite/pull/4674). An oversight maybe?